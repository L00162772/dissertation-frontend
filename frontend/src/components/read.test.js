import React from 'react';
import { shallow } from 'enzyme';
import {render, screen, fireEvent, cleanup} from '@testing-library/react';
import Read from './read';
import {BrowserRouter as Router} from 'react-router-dom';
import axios from 'axios'
jest.mock('axios');

afterEach(cleanup);
afterEach(() => {
    axios.get.mockClear();
  });

  function mockGetDataCallNoData() {
    axios.get.mockResolvedValueOnce({
      data: {}
    });
  }
  function mockGetDataCall1Result() {
    axios.get.mockResolvedValueOnce({
      data: {
        results: [
          {
            id: 1,
            firstName: 'First Name',
            lastName: 'Last Name',
            checkbox: 'Checked'
          },
        ]
      }
    });
  }
  
describe('<Read /> with no props', () => {
  let useEffect;
  const mockUseEffect = () => {
    useEffect.mockImplementationOnce(f => f());
  };
  beforeEach(() => {
    /* mocking useEffect */
    useEffect = jest.spyOn(React, "useEffect");
    mockUseEffect(); // 2 times
    mockUseEffect(); //
  });
  const container = shallow(<Router><Read /></Router>);

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
  
  it('should not load any content', () => {
    render(<Router><Read /></Router>);

    mockGetDataCall1Result()
    //const button = screen.getByTestId('update')
 


    //fireEvent.click(button)
  });
});