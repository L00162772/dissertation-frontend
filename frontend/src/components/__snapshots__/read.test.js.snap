// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Read /> with no props should match the snapshot 1`] = `"<div><table class=\\"ui celled table\\"><thead class=\\"\\"><tr class=\\"\\"><th class=\\"\\">First Name</th><th class=\\"\\">Last Name</th><th class=\\"\\">Checkbox Value</th><th class=\\"\\">Update</th><th class=\\"\\">Delete</th></tr></thead><tbody class=\\"\\"></tbody></table></div>"`;
